### PR TITLE
[WIP] Update search handler to implement changes in kernel forced by EZP-25088

### PR DESCRIPTION
**Pull request by alongosz**
Related to changes made for [JIRA issue EZP-25088](https://jira.ez.no/browse/EZP-25088)
Depends on kernel changes done in [PR #1660](https://github.com/ezsystems/ezpublish-kernel/pull/1660)

It includes:
- [x] Refactored `EzSystems\EzPlatformSolrSearchEngine\FieldValueMapper` namespace to use common field value mappers in `eZ\Publish\Core\Search\Common` namespace
- [x] Implementation of new methods `EzSystems\EzPlatformSolrSearchEngine\Handler::generateDocument` and `EzSystems\EzPlatformSolrSearchEngine\Handler::bulkIndexDocuments` required by new version of `eZ\Publish\SPI\Search\Handler` interface
- [x] Better logging of database inconsistencies while executing `ezplatform:solr_create_index` command (possible due to changes in `eZ\Publish\SPI\Search\Handler`).

### CODE REVIEW

File: bundle/Command/SolrCreateIndexCommand.php [hunk 1]
```diff
@@ -113,4 +115,48 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+    private function generateDocuments(array $contentObjects, SolrSearchEngineHandler $searchHandler, ProgressHelper $progress, OutputInterface $output)
+    {
+        $documents = [];
+
+        foreach ($contentObjects as $content) {
+            try {
+                $documents[] = $searchHandler->generateDocument($content);
+            } catch (NotFoundException $e) {
+                // ignore content objects without assigned state id
+                $progress->clear();
```
Comment (andrerom) [inline bundle/Command/SolrCreateIndexCommand.php]: seems you can move this and `->display()` into `logWarning `

**Comment by andrerom** [comment]
you need to split this into two PR's:
- one where you don't remove anything, you just change handler to account for the interface changes in handler upstream, meaning deprecate `bulkIndexContent ` and add `generateDocuments` and `bulkIndexDocuments` => this change is "soft" BC, it should continue to work on 5.4 and 6.x kernels.
- second PR can be about cleaning up mappers to depend on new features in kernel, this will need to bump version number of SolrBundle to 1.1.

File: bundle/Command/SolrCreateIndexCommand.php (hunk 1 above)
Comment (alongosz) [inline bundle/Command/SolrCreateIndexCommand.php]: Ok

**Comment by alongosz** [comment]
Ok

**Comment by alongosz** [comment]
For now changed to [WIP] as we changed scope of [PR #1660](https://github.com/ezsystems/ezpublish-kernel/pull/1660). I'll get back to this PR when other changes will be completed.

**Comment by alongosz** [comment]
Way better implementation of this can be found in #50 (using common parts introduced in [Kernel PR #1697](https://github.com/ezsystems/ezpublish-kernel/pull/1697)), so I'm closing this PR.

**Pull request closed without merging**